### ISSUE TITLE
Update config.json for nextercism

### ISSUE THREAD

**Issue by ricemery** [exercism/scala#398]
See pull request - #396 .

We need to complete the work started in #396.

**Comment by ErikSchierboom** [comment]
I've have worked on the same issue in the C# and F# track, so here is how I approached this:

1. I selected a set of core exercises that when completed, introduce virtually all relevant concepts of the language to the user. To do this, I just went through the exercises in ascending difficulty order, keeping track of what topics were already introduced by previously selected core exercises.
2. I selected a set of three non-core exercise that will automatically be unlocked when the user submits the first exercise. The exercises should be quite simple.
3. To specify the `unlocked_by` field for the remaining, non-core exercises, I then moved through the core exercises in ascending difficulty. I then searched for first five/six non-core exercises that have overlapping topics with the selected core exercise and marked them as unlocked by the core exercise. Note that this once again assumes that the non-core exercises are in ascending difficulty order.

Having followed these steps, the resulting config should have a set of core exercises, three optional exercises and the remainder is a set of non-core exercises that is unlocked by a core exercise.

Shall we both try to create a list of core exercises?

**Comment by ricemery** [comment]
Your plan sounds good to me. I'll try to come up with what I think should
be the core exercises. Hopefully I'll find some time the next few days.

-r

**Comment by ErikSchierboom** [comment]
My first attempt has the following core exercises:

```json
{
    "uuid": "1fadd9c3-a7c9-4355-aef9-82e0df3d45b9",
    "slug": "hello-world",
    "core": false,
    "unlocked_by": null,
    "difficulty": 1,
    "topics": [
        "Strings"
    ]
},
{
    "uuid": "57b76aba-a485-464d-84ba-e9a445b25be6",
    "slug": "bob",
    "core": false,
    "unlocked_by": null,
    "difficulty": 1,
    "topics": [
        "Strings",
        "Control-flow (if-else statements)",
        "Pattern matching"
    ]
},
{
    "uuid": "7221946d-6321-4cec-8e64-050abae3ccd7",
    "slug": "sum-of-multiples",
    "core": false,
    "unlocked_by": null,
    "difficulty": 1,
    "topics": [
        "Filtering",
        "Sets"
    ]
},
{
    "uuid": "b600cda0-f5b8-45b3-bfd0-7e0c19a0b073",
    "slug": "leap",
    "core": false,
    "unlocked_by": null,
    "difficulty": 1,
    "topics": [
        "Integers",
        "Mathematics"
    ]
},
{
    "uuid": "f147966d-97ec-4479-8679-8cad942b499a",
    "slug": "space-age",
    "core": false,
    "unlocked_by": null,
    "difficulty": 1,
    "topics": [
        "Dates",
        "Enumerations",
        "Domain-specific languages"
    ]
},
{
    "uuid": "b469a197-adf4-4d02-b129-07f993104054",
    "slug": "grade-school",
    "core": false,
    "unlocked_by": null,
    "difficulty": 2,
    "topics": [
        "Maps",
        "Sequences",
        "Sorting"
    ]
},
{
    "uuid": "6d136d51-390a-46a7-b42b-fdecff7b622a",
    "slug": "accumulate",
    "core": false,
    "unlocked_by": null,
    "difficulty": 2,
    "topics": [
        "Generics",
        "Lists",
        "Transforming"
    ]
},
{
    "uuid": "5a31217f-02fd-4be4-b64f-7a93f36f5140",
    "slug": "hamming",
    "core": false,
    "unlocked_by": null,
    "difficulty": 3,
    "topics": [
        "Strings",
        "Filtering",
        "Optional values"
    ]
},
{
    "uuid": "d21f16ce-8e6e-436e-918b-e973cfc7c2a0",
    "slug": "etl",
    "core": false,
    "unlocked_by": null,
    "difficulty": 3,
    "topics": [
        "Lists",
        "Maps",
        "Sequences",
        "Strings",
        "Transforming"
    ]
},
{
    "uuid": "612b8880-1dd1-410d-b530-c66ec9edc1b3",
    "slug": "perfect-numbers",
    "core": false,
    "unlocked_by": null,
    "difficulty": 3,
    "topics": [
        "Discriminated unions",
        "Enumerations",
        "Integers",
        "Mathematics"
    ]
},
{
    "uuid": "e810d2eb-5c90-4135-8db8-0bda54a33d2e",
    "slug": "secret-handshake",
    "core": false,
    "unlocked_by": null,
    "difficulty": 3,
    "topics": [
        "Lists",
        "Strings",
        "Bitwise operations"
    ]
},
{
    "uuid": "261f75c1-df67-4c62-a2e9-ce13550f5de3",
    "slug": "robot-simulator",
    "core": false,
    "unlocked_by": null,
    "difficulty": 3,
    "topics": [
        "Enumerations",
        "Tuples"
    ]
},
{
    "uuid": "c6c37479-a030-44ba-9da5-97eb77233ac6",
    "slug": "matrix",
    "core": false,
    "unlocked_by": null,
    "difficulty": 4,
    "topics": [
        "Strings",
        "Matrices",
        "Parsing",
        "Vectors"
    ]
},
```


These introduce the following topics (in listed order):

- Strings
- Control-flow (if-else statements)
- Pattern matching
- Filtering
- Sets
- Integers
- Mathematics
- Dates
- Enumerations
- Domain-specific languages
- Maps
- Sequences
- Sorting
- Optional values
- Transforming
- Discriminated unions
- Lists
- Bitwise operations
- Tuples
- Matrices
- Parsing
- Vectors

**Comment by ricemery** [comment]
Sorry for the delay with my review of the issue.

I think what you have suggested sounds good. What do you think about adding
"bank-account" as a core exercise - with topic parallelism? I'm not sure,
"bank-account" is the only exercise we would have on that topic.
Similarly, "lens-person" has no topics assigned to it, so I am not sure
were is would fall.

**Comment by ErikSchierboom** [comment]
I'm totally fine with that. I also have no idea where "lens-person" would belong. In the C# track, I had a similar problem with the reactive programming exercises. They don't naturally unlock from any of the core exercises, so I just had them unlock by one of the more difficult core exercises.

**Comment by ricemery** [comment]
I wonder if there should be an "Other" topic for language specific
features. Or, a "Language Specific" topic. For topics like Lens.
Though I recall a discussion about language specific features. I don't
recall the outcome of the discussion. My recollection is that the
collective of maintainers preferred to avoid language tracks having
language specific exercises. Maybe this is less of an issue with Nextercism.

**Comment by ErikSchierboom** [comment]
I thought with Nextercism greater customization of tracks will be the norm. "Other" doesn't really seem right, but "Language Specific" is better.

**Comment by ErikSchierboom** [comment]
This has been fixed in #427. We now have:

- Difficulties assigned to all exercises
- Topics added to the exercises
- Have core/non-core exercises
- Have non-core exercises unlocked by a core exercise
